### ISSUE TITLE
`typeof` on a well-known symbol in a computed property name has a poor error

### ISSUE THREAD

**Issue by DanielRosenwasser** [microsoft/TypeScript#42523]
Found talking to @benlesh

If you try to write the following

```ts
interface SymbolConstructor {
  observable: symbol;
}

declare function from<T>(obj: { [typeof Symbol.observable](): Observable<T> }): Observable<T>;
//                               ^^^^^^ Look at this use of 'typeof'.

type Observer<T> = (x: T) => void;

interface Observable<T> {
  subscribe(observer: Observer<T>): { unsubscribe(): void }
}
```

You'll get an error like

```
A computed property name in a type literal must refer to an expression whose type is a literal type or a 'unique symbol' type.
```

This is pretty confusing, because we *do* support well-known symbols that are declared with `symbol` instead of unique symbol if you use the following syntax:

```
declare function from<T>(obj: { [Symbol.observable](): Observable<T> }): Observable<T>;
//                               ^ No 'typeof'.
```

It would be great if we could detect this case and specialize the error message with something like

```
Computed property names can only be named with well-known symbols without the 'typeof' operator. Try replacing `typeof {0}` with `{0}`.
```

**Comment by jonhue** [comment]
#42530 adds this error message 🙂

**Comment by benlesh** [comment]
Related... this also has issues (given any permutation of your example code):

```ts
// Imagine you got a reference to the symbol from somewhere else.
const symbolObservable = Symbol.observable;

const obj = {
    [symbolObservable](): Observable<any> {
        return null!;
    }
}

const x = from(obj);
```

**Comment by DanielRosenwasser** [comment]
This is really just going to be closed by https://github.com/microsoft/TypeScript/pull/24738 if I understand correctly.

**Comment by DanielRosenwasser** [comment]
Or I guess https://github.com/microsoft/TypeScript/pull/42543 now

**Comment by jonhue** [comment]
@DanielRosenwasser just out of curiosity: was this fixed by #42543 and could be closed or should this remain open? 🙂

**Comment by DanielRosenwasser** [comment]
Yup, thanks!
